src/pixi-multistyle-text.ts

```diff
@@ -148,768 +196,809 @@ interface TextWithPrivateMembers extends PIXI.Text {
+    letterSpacing: 0,
+    lineHeight: 0,
+    lineJoin: "miter",
+    miterLimit: 10,
+    padding: 0,
+    stroke: "black",
+    strokeThickness: 0,
+    textBaseline: "alphabetic",
+    wordWrap: false,
+    wordWrapWidth: 100,
+
+    tagStyle: "xml",
+    debug:false,
+    valign: "baseline",
+  };
+
+  public static debugOptions: MstDebugOptions = {
+    spans: {
+      enabled: false,
+      baseline: "#44BB44",
+      top: "#BB4444",
+      bottom: "#4444BB",
+      bounding: "rgba(255, 255, 255, 0.1)",
+      text: true
+    },
+    objects: {
+      enabled: false,
+      bounding: "rgba(255, 255, 255, 0.05)",
+      text: true
+    }
+  };
+
+  private _textStyles!: TextStyleSet;
+
+  public get textStyles () { return this._textStyles; }
+
+  public set textStyles(styles: TextStyleSet) {
+    throw new Error("Don't set textStyles directly. Use setStyles()");
+  }
+
+  public get defaultTextStyle () { return this.textStyles.default; }
+  public set defaultTextStyle (style) { this.textStyles.default = style; }
+
+  public setStyles(styles:TextStyleSet):void {
+    this.resetTextStyles();
+
+    for (let styleName in styles) {
+      if (styleName === "default") {
+        this._textStyles.default = {...this._textStyles.default, ...styles.default};
+      } else {
+        this._textStyles[styleName] = { ...styles[styleName] };
+      }
+    }
+    if (this._textStyles.default.tagStyle === TAG_STYLE.bbcode) {
+      // when using bbcode parsing, register a bunch of standard bbcode tags and some cool pixi ones
+      this._textStyles.b = {fontStyle: 'bold'};
+      this._textStyles.i = {fontStyle: 'italic'};
+      this._textStyles.color = {fill: ''}; // an array would result in gradiens
+      this._textStyles.outline = {stroke: '', strokeThickness: 6};
+      this._textStyles.font = {fontFamily: ''};
+      this._textStyles.shadow ={ dropShadowColor: '', dropShadow: true, dropShadowBlur: 3, dropShadowDistance: 3, dropShadowAngle: 2};
+      this._textStyles.size = {fontSize: 'px'};
+      this._textStyles.spacing = {letterSpacing: 0};
+      this._textStyles.align = {align: ''};
+    }
+
+    this.withPrivateMembers()._style = new PIXI.TextStyle(this._textStyles.default);
+    this.withPrivateMembers().dirty = true;
+  }
+
+  private resetTextStyles() {
+    this._textStyles = { default : {...MultiStyleText.DEFAULT_TAG_STYLE} };
+  }
+
+  private hitboxes: HitboxData[] = [];
+
+  constructor(text: string, styles: TextStyleSet) {
+    super(text);
+
+    this.setStyles(styles);
+
+    const migrateEvent = (e: PIXI.interaction.InteractionEvent) => this.handleInteraction(e);
+
+    INTERACTION_EVENTS.forEach((event) => {
+      this.on(event, migrateEvent);
+    });
+  }
+
+  private handleInteraction(e: PIXI.interaction.InteractionEvent) {
+    let ev = e as MstInteractionEvent;
+
+    let localPoint = e.data.getLocalPosition(this);
+    let targetTag = this.hitboxes.reduce(
+      (prev:HitboxData | undefined, hitbox:HitboxData) => {
```

Comment:
You can move the type to `reduce<HitboxData | undefined>` and it should work the same as extra typing the params. Same for `hitbox`, we shouldn't need to be explicit.

Based on the content of the reduce function, it looks like we should use `find` instead?